chapters.lua

```diff
@@ -537,11 +537,12 @@ end
 --      format - string for file extension and content
 --      osd - if true, display an osd message
 --      store - if true store chapters according to the config file
+--      manual_save - if true force a save even without chapter modifications
 --      chapter_zero - if true generate a chapter at 0:00 if one doesn't exists
 -- on success returns path of the chapters file, nil on failure
 local function write(...)
-    local format, osd, store, chapter_zero = ...
-    if store and not chapters_modified then
+    local format, osd, store, manual_save, chapter_zero = ...
+    if (store and not chapters_modified) and not manual_save then
```

Comment:
```suggestion
    if not chapters_modified and not manual_save then
```
shouldn't that be the condition? if you still check for `store` then the calls to `write` from `autoexportformat` hook will always fail this condition

Reply:
Hm that is a good point, but I did test that and I did get an automatic export... but I'm now questioning *why* that works :thinking:

Reply:
> This is supposed to retain the current behaviour of writing nothing if `store` is set and nothing was modified.

So what is the point of the flag if it doesn't change anything? I thought the idea was to avoid `write` from `autoexportformats` hook when there are no chapter modifications, no? As it is, it doesn't do that.

Reply:
> So what is the point of the flag if it doesn't change anything? I thought the idea was to avoid `write` from `autoexportformats` hook when there are no chapter modifications, no? As it is, it doesn't do that.

Yeah you're right... I might be too sleepy right now.

When writing this I was only looking at the manual save aspect but sort of forgot about the autoexport when nothing was changed.
I'll have to take a look again after taking a nap :sweat_smile:

Reply:
OK, I adjusted it, but while testing `bake_chapters` something came up with it.

So the current behaviour even when nothing was changed is that it writes the `.tmp` chapter file, writes the new "baked" video file, and then deletes the `.tmp`.
That's because `bake_chapters()` calls `write()` with `store = false`:
https://github.com/mar04/chapters_for_mpv/blob/ed7f63a5dbbad9302c4af2be347a7545954a14ec/chapters.lua#L707
Which of course fails and doesn't return early.

With this PR, the chapter file isn't written and `bake_chapters()` returns early because of it, because there's nothing to bake:
https://github.com/mar04/chapters_for_mpv/blob/ed7f63a5dbbad9302c4af2be347a7545954a14ec/chapters.lua#L708-L711

So the question is... which behaviour is preferred here? Should the current behaviour be retained? We can do that by passing `manual_save = true`.
But I'd argue not writing anything is actually... sort of a bug fix here? Baking chapters when nothing is modified is pretty pointless and depending on the video might result in a huge write operation for something that is basically a copy of the existing file.

Reply:
Embedding an empty chapters file is of little use, I'm fine with returning early with a better message.

Reply:
No, scratch that. On a second thought, I think it's better to keep the current behavior. So pass `manual_save=true` in those calls. `chapters_modified` doesn't track changes across different times you open the file, so even if it's false at a time we call `bake`, it's not guaranteed that the chapters are the same as the ones embedded in the file.

Reply:
A right, didn't think about re-opening the file.
Fixed up.